playbooks/roles/airship-setup-deployer/tasks/main.yml

```diff
@@ -116,3 +116,10 @@
   template:
     src: clouds.yml.j2
     dest: /etc/openstack/clouds.yaml
+
+- name: Save Shipyard env variables to ~/.profile
+  blockinfile:
+    path: "{{ ansible_user_dir }}/.profile"
```

Comment:
I think `blockinfile` will not create the file if it doesn't exist yet by default. Also I think we should not be putting passwords into `.profile` (it often is word-readable). How about creating some `.airshiprc`  (properly protected)? And either source that from `.profile` or drop a file into `/etc/profile.d` that sources it if it exists. Or not source it at all automatically and leave it as an excerise for the user to do so.

I know that this is a bit moot as long as the deployer is still some well separated stand-alone node. But this might change in the future.

Reply:
I like e.g. `/etc/profile.d/airship.sh`. But that would make it available for all users, right?

Reply:
So `.airshiprc` in `"{{ ansible_user_dir }}` seems to be safest bet I think

Reply:
And with custom file like this I could go back to using template instead of `blockinfile`

Reply:
> I like e.g. `/etc/profile.d/airship.sh`. But that would make it available for all users, right?

Well. In theory that file could just source `~/.airshiprc` if it exists, the it would keep the password "secret" and only work for users that actually have that file. But I am rather unsure if that is really needed at the current stage.

> So `.airshiprc` in `"{{ ansible_user_dir }}` seems to be safest bet I think

Yes

> And with custom file like this I could go back to using template instead of `blockinfile`

Yes. (Note though, that `blockinfile` has an option that allows to create the file if it doesn't exist)

Reply:
OK, done (with blockinfile as I was lazy to look up the template again)